File: csharp/exercises/Webshop.Service/Orders/OrderRepository.cs
namespace Webshop.Service.Orders;

public interface IOrderRepository
{
    Order AddOrder(string productId, int quantity, string userId);
}

public class OrderRepository : IOrderRepository
{
    private Dictionary<string, Order> _orderDatabase = new();

    public Order AddOrder(string productId, int quantity, string userId)
    {
        var order = new Order { OrderId = Guid.NewGuid().ToString(), ProductId = productId, Quantity = quantity, UserId = userId };
        _orderDatabase.Add(order.OrderId, order);
        return order;
    }
}

public class Order
{
    public string OrderId { get; set; }
    public string ProductId { get; set; }
    public int Quantity { get; set; }
    public string UserId { get; set; }
}

File: java/exercises/src/main/java/Webshop/Service/Orders/OrderRepository.java
package java.Webshop.Service.Orders;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

interface IOrderRepository {
    Order addOrder(String productId, int quantity, String userId);
}

public class OrderRepository implements IOrderRepository {
    private final Map<String, Order> orderDatabase = new HashMap<>();

    @Override
    public Order addOrder(String productId, int quantity, String userId) {
        Order order = new Order(UUID.randomUUID().toString(), productId, quantity, userId);
        orderDatabase.put(order.getOrderId(), order);
        return order;
    }
}

class Order {
    private String orderId;
    private String productId;
    private int quantity;
    private String userId;

    public Order(String orderId, String productId, int quantity, String userId) {
        this.orderId = orderId;
        this.productId = productId;
        this.quantity = quantity;
        this.userId = userId;
    }

    public String getOrderId() {
        return orderId;
    }

    public String getProductId() {
        return productId;
    }

    public int getQuantity() {
        return quantity;
    }

    public String getUserId() {
        return userId;
    }
}

